### PR TITLE
Feature/add review entity v2

### PR DESCRIPTION
**Pull request by WWOWNOW**
v2

### CODE REVIEW

**Review by Sunagatov — APPROVED** [review]

File: src/main/java/com/zufar/onlinestore/review/controller/ReviewController.java [hunk 1]
```diff
@@ -71,7 +71,8 @@ public ResponseEntity<ApiResponse<String>> deleteReview(@PathVariable String rev
         log.info("The Review with id - {} was deleted.", reviewId);
 
         ApiResponse<String> apiResponse = ApiResponse.<String>builder()
-                .message("Review successfully deleted")
+                .data(reviewId)
+                .message("The Review with id - " + reviewId + " was deleted.")
```
[the remark this answers is not in the document]
Comment (WWOWNOW) [reply src/main/java/com/zufar/onlinestore/review/controller/ReviewController.java:74]: fixed

File: src/main/java/com/zufar/onlinestore/review/controller/ReviewController.java [hunk 2]
```diff
@@ -89,16 +90,17 @@ public ResponseEntity<ApiResponse<ReviewDto>> editReview(@PathVariable String re
...
+
+        ReviewDto updatedReviewDto = reviewService.addReview(existingReviewDto);
+
+        log.info("The review with id - {} was edited.", updatedReviewDto.getId());
 
         ApiResponse<ReviewDto> apiResponse = ApiResponse.<ReviewDto>builder()
-                .data(existingReviewDto)
-                .message("The review with id " + existingReviewDto.getId() + " was edited.")
+                .data(updatedReviewDto)
+                .message("The review with id " + updatedReviewDto.getId() + " was edited.")
```
[the remark this answers is not in the document]
Comment (WWOWNOW) [reply src/main/java/com/zufar/onlinestore/review/controller/ReviewController.java:99]: fixed

File: src/main/java/com/zufar/onlinestore/review/service/ReviewService.java [hunk 3]
```diff
@@ -38,6 +40,11 @@ public List<ReviewDto> getProductReviews(String productId) {
     public String deleteReview(String reviewId) {
         reviewRepository.findById(reviewId).orElseThrow(() -> new ReviewNotFoundException(reviewId));
```
[the remark this answers is not in the document]
Comment (WWOWNOW) [reply src/main/java/com/zufar/onlinestore/review/service/ReviewService.java:41]: fixed

File: src/main/java/com/zufar/onlinestore/review/service/ReviewService.java [hunk 4]
```diff
@@ -27,7 +27,9 @@ public ReviewDto addReview(ReviewDto reviewDto) {
     }
 
     public Optional<ReviewDto> findReview(String reviewId) {
-        return reviewRepository.findById(reviewId).map(reviewDtoConverter::convertToDto);
+        return Optional.ofNullable(reviewRepository.findById(reviewId)
+                .map(reviewDtoConverter::convertToDto)
+                .orElseThrow(() -> new ReviewNotFoundException(reviewId)));
```
[the remark this answers is not in the document]
Comment (WWOWNOW) [reply src/main/java/com/zufar/onlinestore/review/service/ReviewService.java:32]: fixed

**Pull request merged**